beef/bump.go

```diff
@@ -56,23 +57,34 @@ func calculateMerkleRoot(baseLeaf BUMPLeaf, bump BUMP) (string, error) {
 	calculatedHash := baseLeaf.Hash
 	offset := baseLeaf.Offset
 
-	for _, bLevel := range bump.Path {
+	for i := 0; i < len(bump.Path); i++ {
+		bLevel := bump.Path[i]
+		var previousBLevel []BUMPLeaf
+
+		if i-1 >= 0 {
+			previousBLevel = bump.Path[i-1]
+		}
 		newOffset := getOffsetPair(offset)
 		leafInPair := findLeafByOffset(newOffset, bLevel)
 		if leafInPair == nil {
-			return "", errors.New("could not find pair")
+			var err error
+			if previousBLevel != nil {
+				leafInPair, err = calculateFromChildren(newOffset, previousBLevel)
+				if err != nil {
+					return "", err
+				}
+			} else {
+				return "", errors.New("cannot compute leaf from children at base level")
+			}
 		}
```

Comment:
I made a small adjustment to the code structure, and after running spv_test, everything seems to be working as expected 😄. 

```
if leafInPair == nil {
	if previousBLevel == nil {
		return "", errors.New("cannot compute leaf from children at base level")
	}
			
	var err error
	leafInPair, err = calculateFromChildren(newOffset, previousBLevel)
	if err != nil {
		return "", err
	}
}
```

Reply:
fixed